mappings/net/minecraft/util/math/NoiseInterpolator.mapping

```diff
@@ -0,0 +1,57 @@
+CLASS net/minecraft/class_5917 net/minecraft/util/math/NoiseInterpolator
+	FIELD field_29227 startNoiseBuffer [[D
+	FIELD field_29228 endNoiseBuffer [[D
+	FIELD field_29229 sizeY I
+	FIELD field_29230 sizeZ I
+	FIELD field_29231 minY I
+	FIELD field_29232 columnSampler Lnet/minecraft/class_5917$class_5918;
+	FIELD field_29233 noise000 D
```

Comment:
Personally, I'm much more partial to the name `x0z0y0` instead of `noise000` for example. It's easier to see what each of the 3 numbers mean. Interested to hear what you all have to say

Reply:
Just for reference, in a similar situation in matrix4f, we currently name fields like a00, a01, etc. Not saying that naming is the best, but it'd help if we can find a consistent reasoning for naming fields in different styles than adding 15 competing standards.

Reply:
That's a good point. I'm mostly interested in naming it this way as the values get interpolated in the 3 axes separately, so it's easier to see how it occurs, e.g. when the cube gets interpolated down to a square, the y value is "lost" as it's interpolated into the x and z values, so `x0z0y0` becomes `x0z0` for example. The same happens when the square is interpolated into a line, where it becomes `z0` from `x0z0` as the x value is "lost" and interpolated into the z values.

Reply:
Yeah, we can choose to use `xz00` names, which has more specific prefixes (about the type of lerping they are doing) than the generic "noise", already implied by the class name "noise interpolator".

Imo we should pull out `xz` `xyz` as a whole to the prefix, like `xyz000` than `x0y0z0` so in code it's be more obvious which types of lerping arguments you are using.

Reply:
I originally went with the numbers interlayed with the coord as it was originally how the noise was accessed out of the 3d array, so it was:
```java
double x0z0y0 = noiseData[0][noiseZ][noiseY];
double x0z1y0 = noiseData[0][noiseZ + 1][noiseY];
double x1z0y0 = noiseData[1][noiseZ][noiseY];
double x1z1y0 = noiseData[1][noiseZ + 1][noiseY];
```
so it was x first, z second, and y third. Since these are local variables it becomes a bit hairier, but I still think we should keep the interlayed coord as it's easier to me to see which axis it is interpolating. Doing the prefix would lose the meaning of `x0z0y0` meaning "x0, z0, y0" in my opinion.

Reply:
Agreed. If `x0z0y0` is better than `xzy000` for understanding code, I'm all for it. The purpose of yarn is to allow us understand minecraft code better (say, than if we look at mojang map)

Reply:
I named the fields that way because of `MathHelper.lerp3`. But you're right, `noise000` is meaningless on its own.

Reply:
Who was that idiot who named the parameters in `lerp3`? Oh, darn, it was me. Tbh I couldn't think of any better parameter names at the time.
Anyway, the naming inside mat4f is pretty standard specifically for matrices, but it may be confusing outside of that context.

Reply:
> Who was that idiot who named the parameters in lerp3? Oh, darn, it was me. Tbh I couldn't think of any better parameter names at the time.

They were at least better than the names I originally came up for this.